src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/SuppressionEngine.cs

```diff
@@ -30,7 +30,10 @@ public class SuppressionEngine : ISuppressionEngine
             string? noWarn = null,
             bool baselineAllErrors = false)
         {
-            _validationSuppressions = ParseSuppressionFile(suppressionFile);
+            // Don't read from the passed in suppression file if errors will be baselined and the file doesn't exist.
+            _validationSuppressions = baselineAllErrors && !File.Exists(suppressionFile) ?
```

Comment:
>_validationSuppressions [](http://example.com/codeflow?start=12&length=23)

Even if the file does exist, if baselineAllErrors is true technically we could also skip the parsing of the file right?
Or is the problem there that if we do that we would expand global suppressions in that case?

Reply:
To me, it seems like

_validationSuppressions = new HashSet<Suppression>();

try
{
    ParseSuppressionFile(suppressionFile, _validationSuppressions);
}
catch (FileNotFoundException) when (baselineAllErrors)
{
}
```

is a better fix, since it avoids the trap of using File.Exists.

Reply:
> Even if the file does exist, if baselineAllErrors is true technically we could also skip the parsing of the file right?
Or is the problem there that if we do that we would expand global suppressions in that case?

If the suppression file exists but isn't read, global suppressions in it would be lost. [This logic was already part of APICompat v1](https://github.com/dotnet/sdk/blob/79a92b4eb087b07494034d0a09168dcf323f1efe/src/Compatibility/Microsoft.DotNet.Compatibility/CompatibilityLogger.cs#L24) but was lost during the addition of the CLI frontend.

I noticed this just last week when I passed in a suppression-file option to the CLI frontend which resulted in an uncaught FileNotFoundException as the file didn't exist.